jmespath-jakarta-jsonp/src/main/java/io/burt/jmespath/jakarta/jsonp/JsonpRuntime.java

```diff
@@ -0,0 +1,230 @@
+  }
+
+  public JsonpRuntime(RuntimeConfiguration configuration, JsonReaderFactory jsonReaderFactory) {
+    super(configuration);
+    this.jsonReaderFactory = jsonReaderFactory;
+  }
+
+  @Override
+  public JsonValue parseString(String string) {
+    return jsonReaderFactory.createReader(new StringReader(string)).readValue();
+  }
+
+  private static class JsonArrayListWrapper extends AbstractList<JsonValue> {
+    private final JsonArray array;
+
+    JsonArrayListWrapper(JsonArray array) {
+      this.array = array;
+    }
+
+    @Override
+    public JsonValue get(int index) {
+      return array.get(index);
+    }
+
+    @Override
+    public int size() {
+      return array.size();
+    }
+  }
+
+  @Override
+  public List<JsonValue> toList(JsonValue value) {
+    ValueType valueType = value.getValueType();
+    if (valueType == ARRAY) {
+      return new JsonArrayListWrapper((JsonArray) value);
+    } else if (valueType == OBJECT) {
+      JsonObject obj = (JsonObject)value;
+      if (!obj.isEmpty()) {
+        List<JsonValue> elements = new ArrayList<>(obj.size());
+        for (JsonValue v : obj.values()) {
+          elements.add(useJsonNull(v));
+        }
+        return elements;
+      }
+    }
+    return Collections.emptyList();
+  }
+
+  @Override
+  public String toString(JsonValue str) {
+    if (str.getValueType() == STRING) {
+      return ((JsonString)str).getString();
+    } else {
+      return str.toString();
+    }
+  }
+
+  @Override
+  public Number toNumber(JsonValue n) {
+    return (n.getValueType() == NUMBER) ? ((JsonNumber)n).numberValue() : null;
+  }
+
+  @Override
+  public boolean isTruthy(JsonValue value) {
+    switch (value.getValueType()) {
+      case FALSE:
+      case NULL:
+        return false;
+      case NUMBER:
+      case TRUE:
+        return true;
+      case ARRAY:
+        return ((JsonArray)value).size() > 0;
+      case OBJECT:
+        return ((JsonObject)value).size() > 0;
+      case STRING:
+        return ((JsonString)value).getString().length() > 0;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JmesPathType typeOf(JsonValue value) {
+    switch (value.getValueType()) {
+      case ARRAY:
+        return JmesPathType.ARRAY;
+      case OBJECT:
+        return JmesPathType.OBJECT;
+      case STRING:
+        return JmesPathType.STRING;
+      case FALSE:
+      case TRUE:
+        return JmesPathType.BOOLEAN;
+      case NULL:
+        return JmesPathType.NULL;
+      case NUMBER:
+        return JmesPathType.NUMBER;
+      default:
+        throw new IllegalStateException(String.format("Unknown node type encountered: %s", value.getValueType()));
+    }
+  }
+
+  @Override
+  public JsonValue getProperty(JsonValue value, JsonValue name) {
+    if (value.getValueType() == OBJECT) {
+      return nodeOrNullNode(((JsonObject)value).get(textOnStringOrNull(name)));
```

Comment:
How come the call to `textOnStringOrNull` here? The object can't have a null key. I think `name.getString()` should be sufficient here.

Reply:
>  it's very nice to see how it's possible to support another library with this little effort.

Indeed! It just took some hours (part of a day) and most of the effort was spent figuring out the right semantics of this API.

... and I do not benefit from that much experience with JSON, which kind of shows on the glitches you pointed out.

The test battery helps a lot!!!

Reply:
Congratulations. This library is a really easy both to use and to code for!
:)